plugins/github-plugin/parsegitlog.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/local/bin/python3
+
+import json
+import os
+from collections import OrderedDict
+
+import git
+import requests
+from kafka import KafkaConsumer
+import configparser
+
+# TODO manage the kafka offset for restarts, error handling
+config = configparser.ConfigParser()
+config.read('config.ini')
+consumer = KafkaConsumer(config['Kafka']['topic'],
+                         bootstrap_servers=[config['Kafka']['url'] + ":" + config['Kafka']['port']])
+url = 'http://' + config['Ground']['url'] + ':' + config['Ground']['port']
+
+for msg in consumer:
+
+    parsed_msg = json.loads(msg.value.decode("utf-8"))  # turn into json
+
+    print(msg.key.decode("utf-8"))  # print key
+
+    gitUrl = parsed_msg['repository']['git_url']
+    # gitUrl = 'git://github.kdc.capitalone.com/gryzzl/edt.git'
+    print(gitUrl)
+    # repoId = msg.key.decode("utf-8")  # key from kakfa
+    repoId = str(parsed_msg['repository']['id'])
+
+    if not os.path.exists(config['Git']['path'] + repoId):
+        repo = git.Repo.init(config['Git']['path'] + repoId, bare=True)
+        origin = repo.create_remote('origin', url=gitUrl)
+    else:
+        repo = git.Repo(config['Git']['path'] + repoId)
+
+    repo.remotes.origin.fetch()  # update repo metadata
+    g = git.Git(config['Git']['path'] + repoId)
+
+    # URLs
+    nodeUrl = url + '/nodes/' + repoId
+    nodeVUrl = url + '/nodes/versions/'
+    latestUrl = nodeUrl + '/latest'
+
+    parentsLog = g.log('--all', '--format="%H,%P"')  # log with commit hash and parent hashes
```

Comment:
I'm using [gitpython](https://github.com/gitpython-developers/GitPython) right now. The other option is to use `subprocess.run` to run commands like `git rev-list` and return the output. I can't find a way to get the original branch name without using `git log --source` though. Log isn't a plumbing command, but since the library handles it it returns a string.